gamestonk_terminal/mutual_funds/mutual_fund_controller.py

```diff
@@ -0,0 +1,546 @@
+
+import argparse
+import difflib
+from datetime import datetime, timedelta
+from typing import List, Union
+
+import investpy
+import pandas as pd
+from prompt_toolkit.completion import NestedCompleter
+from rich import console
+
+from gamestonk_terminal import feature_flags as gtff
+from gamestonk_terminal.helper_funcs import (
+    EXPORT_ONLY_FIGURES_ALLOWED,
+    EXPORT_ONLY_RAW_DATA_ALLOWED,
+    check_positive,
+    get_flair,
+    parse_known_args_and_warn,
+    system_clear,
+    try_except,
+    valid_date,
+)
+from gamestonk_terminal.menu import session
+from gamestonk_terminal.mutual_funds import investpy_model, investpy_view, yfinance_view
+
+console = console.Console()
+
+
+class FundController:
+    """Fund Controller class"""
+
+    CHOICES = [
+        "cls",
+        "home",
+        "h",
+        "?",
+        "help",
+        "q",
+        "quit",
+        "..",
+        "exit",
+        "r",
+        "reset",
+    ]
+    CHOICES_COMMANDS = [
+        "country",
+        "search",
+        "overview",
+        "info",
+        "load",
+        "plot",
+        "sector",
+        "equity",
+    ]
+
+    CHOICES += CHOICES_COMMANDS
+    fund_countries = investpy.funds.get_fund_countries()
+    search_by_choices = ["name", "issuer", "isin", "symbol"]
+    search_cols = [
+        "country",
+        "name",
+        "symbol",
+        "issuer",
+        "isin",
+        "asset_class",
+        "currency",
+        "underlying",
+    ]
+
+    def __init__(self, queue: List[str] = None):
+        """Constructor"""
+        self.fund_parser = argparse.ArgumentParser(add_help=False, prog="fund")
+        self.fund_parser.add_argument("cmd", choices=self.CHOICES)
+        self.country = "united states"
+        self.completer: Union[None, NestedCompleter] = None
+        self.data = pd.DataFrame()
+        self.fund_name = ""
+        self.fund_symbol = ""
+        if session and gtff.USE_PROMPT_TOOLKIT:
+            choices: dict = {c: {} for c in self.CHOICES}
+            choices["country"]["-n"] = {c: None for c in self.fund_countries}
+            choices["search"]["-b"] = {c: None for c in self.search_by_choices}
+            choices["search"]["--by"] = {c: None for c in self.search_by_choices}
+            choices["search"]["-s"] = {c: None for c in self.search_cols}
+            choices["search"]["--sortby"] = {c: None for c in self.search_cols}
+            self.completer = NestedCompleter.from_nested_dict(choices)
+
+        if queue:
+            self.queue = queue
+        else:
+            self.queue = list()
+
+    def print_help(self):
+        """Print help"""
+        fund_string = f"{self.fund_name or None}"
+        fund_string2 = f" ({self.fund_symbol})" if self.fund_symbol else ""
+        fund_string += fund_string2
+        help_str = f"""
+[bold]Mutual Funds[/bold]:
+
+Current Fund:[cyan] {fund_string}[/cyan]
+Current Country: [green]{self.country.title()}[/green][italic]
+
+Investing.com:[/italic]
```

Comment:
I find this menu a bit confusing, can we have it like this:

```
Investing.com:
    overview    overview of top funds by country
    search        search for Mutual Funds
    load            load historical fund data

Current Fund: Vanguard International Shares Index Fund (0P00007962)

Investing.com:
    info          get fund information
    plot          plot loaded historical fund data

Yahoo Finance:
    sector        sector weightings
    equity        equity holdings
```

Basically the overview having as argument the country makes more sense. UNLESS, you explicitly need the country to load the data.

Reply:
For whatever reason, the country does need to be specified :(

Reply:
Got it, so I would reflect that on the display of the menu, having something like: 

```
Investing.com:
    country       set a country for filtering
 
Current Country: United States

Investing.com:
    overview    overview of top funds by country
    search        search for Mutual Funds
    load            load historical fund data

Current Fund: Vanguard International Shares Index Fund (0P00007962)

Investing.com:
    info          get fund information
    plot          plot loaded historical fund data

Yahoo Finance:
    sector        sector weightings
    equity        equity holdings
```